Security/How to Configure a Service Fabric Managed Cluster with Common Name Certificate.md

```diff
@@ -0,0 +1,259 @@
+
+> ### :exclamation:NOTE: Connecting to a managed cluster endpoint, for example Service Fabric Explorer (SFX), a certificate error (NET::ERR_CERT_AUTHORITY_INVALID) will occur regardless of certificate being used or configuration due to the cluster using a managed 'cluster' certificate.
+
+## Prerequisites
+
+- A Certificate Authority (CA) signed certificate with appropriate SubjectName.
+- Azure Service Fabric SDK which contains PowerShell 'servicefabric' module. This is available on development machines with Visual Studio or on any active node in a service fabric cluster.
+- PowerShell Azure 'Az' modules
+
+## Adding common name certificate as a 'client' certificate for cluster connectivity
+
+Use one of the three options below  to add a common name certificate as a 'client' certificate for cluster connectivity.
+
+### Using Azure Resource Explorer to add common name certificate configuration
+
+- Open https://resources.azure.com and navigate to {{subscription id}}/resourceGroups/{{resource group}}/providers/Microsoft.ServiceFabric/managedClusters/{{cluster name}}
+- Select 'Read/Write' and 'Edit'
+- Populate provided 'clients' new element template
+  - isAdmin - set to true if certificate should have cluster write / management capabilities else set to false for readonly.
+  - commonName - certificate 'SubjectName' without the 'CN='
+  - issuerThumbprint - thumbprint of the Issuing certificate for the common name certificate.
+
+  ```json
+  //"clients": [
+    {
+      "isAdmin": "(Boolean)",
+      "commonName": "(String)",
+      "issuerThumbprint": "(String)"
+    }
+  //]
+  ```
+
+- 'PUT' to update configuration.
+
+### Using PowerShell to add common name certificate configuration
+
+Run the commands below to use PowerShell to update the certificate configuration of the cluster.
+
+Variables:
+
+- resourceGroupName - Azure resource group that contains the managed cluster.
+- commonName - certificate 'SubjectName' without the 'CN='.
+- issuerThumbprint - thumbprint of the Issuing certificate for the common name certificate.
+- admin - set to true if certificate should have cluster write / management capabilities else set to false for readonly.
+
+```powershell
+$resourceGroupName = ''
+$clusterName = $resourceGroupName
+$commonName = '*.sfcluster.com'
+$issuerThumbprint = ''
+$admin = $false
+Add-AzServiceFabricManagedClusterClientCertificate -ResourceGroupName $resourceGroupName `
+   -Name $clusterName `
+   -CommonName $commonName `
+   -IssuerThumbprint $issuerThumbprint `
+   -Admin:$admin
+```
+
+### Using ARM template to add common name certificate to configuration
```

Comment:
Can you reorder this so ARM example is listed first , and label it as the recommended/preferred method?

Reply:
updated